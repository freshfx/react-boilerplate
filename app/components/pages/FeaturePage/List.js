import styled from '@emotion/styled'

const List = styled.ul`
  font-family: Georgia, Times, 'Times New Roman', serif;
  padding-left: 1.75em;
`

export default List
