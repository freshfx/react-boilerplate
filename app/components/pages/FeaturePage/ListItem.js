import styled from '@emotion/styled'

const ListItem = styled.li`
  margin: 1em 0;
`

export default ListItem
