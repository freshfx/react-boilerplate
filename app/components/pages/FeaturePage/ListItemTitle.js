import styled from '@emotion/styled'

const ListItemTitle = styled.p`
  font-weight: bold;
`

export default ListItemTitle
