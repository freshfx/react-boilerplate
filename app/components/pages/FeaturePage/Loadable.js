/**
 * Asynchronously loads the component for FeaturePage
 */
import React from 'react'

import PageLoadingIndicator from 'components/molecules/PageLoadingIndicator'
import withLoadable from 'utils/withLoadable'

export default withLoadable(() => import('./index'), {
  fallback: <PageLoadingIndicator />
})
