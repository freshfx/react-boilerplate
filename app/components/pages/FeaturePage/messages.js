/*
 * FeaturePage Messages
 *
 * This contains all the text for the FeaturePage component.
 */
import {defineMessages} from 'react-intl'

const scope = 'boilerplate.containers.FeaturePage'

export default defineMessages({
  cssHeader: {
    defaultMessage: 'Features',
    id: `${scope}.css.header`
  },
  cssMessage: {
    defaultMessage: 'Next generation CSS',
    id: `${scope}.css.message`
  },
  feedbackHeader: {
    defaultMessage: 'Instant feedback',
    id: `${scope}.feedback.header`
  },
  feedbackMessage: {
    defaultMessage: `
      Enjoy the best DX and code your app at the speed of thought! Your
    saved changes to the CSS and JS are reflected instantaneously
    without refreshing the page. Preserve application state even when
    you update something in the underlying code!
    `,
    id: `${scope}.feedback.message`
  },
  header: {
    defaultMessage: 'Features',
    id: `${scope}.header`
  },
  intlHeader: {
    defaultMessage:
      'Complete i18n Standard Internationalization & Pluralization',
    id: `${scope}.internationalization.header`
  },
  intlMessage: {
    defaultMessage:
      'Scalable apps need to support multiple languages, easily add and support multiple languages with `react-intl`.',
    id: `${scope}.internationalization.message`
  },
  javascriptHeader: {
    defaultMessage: 'Next generation JavaScript',
    id: `${scope}.javascript.header`
  },
  javascriptMessage: {
    defaultMessage: `Use template strings, object destructuring, arrow functions, JSX
    syntax and more, today.`,
    id: `${scope}.javascript.message`
  },
  networkHeader: {
    defaultMessage: 'Offline-first',
    id: `${scope}.network.header`
  },
  networkMessage: {
    defaultMessage: `
      The next frontier in performant web apps: availability without a
      network connection from the instant your users load the app.
    `,
    id: `${scope}.network.message`
  },
  routingHeader: {
    defaultMessage: 'Industry-standard routing',
    id: `${scope}.routing.header`
  },
  routingMessage: {
    defaultMessage: `
      Write composable CSS that's co-located with your components for
    complete modularity. Unique generated class names keep the
    specificity low while eliminating style clashes. Ship only the
    styles that are on the page for the best performance.
    `,
    id: `${scope}.routing.message`
  },
  scaffoldingHeader: {
    defaultMessage: 'Quick scaffolding',
    id: `${scope}.scaffolding.header`
  },
  scaffoldingMessage: {
    defaultMessage: `Automate the creation of components, containers, routes, selectors
  and sagas - and their tests - right from the CLI!`,
    id: `${scope}.scaffolding.message`
  },
  stateManagementHeader: {
    defaultMessage: 'Predictable state management',
    id: `${scope}.state_management.header`
  },
  stateManagementMessages: {
    defaultMessage: `
      Unidirectional data flow allows for change logging and time travel
    debugging.
    `,
    id: `${scope}.state_management.message`
  }
})
