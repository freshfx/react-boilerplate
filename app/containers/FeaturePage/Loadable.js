/**
 * Asynchronously loads the component for FeaturePage
 */
import React from 'react'

import LoadingIndicator from 'components/LoadingIndicator'
import withLoadable from 'utils/withLoadable'

export default withLoadable(() => import('./index'), {
  fallback: <LoadingIndicator />
})
