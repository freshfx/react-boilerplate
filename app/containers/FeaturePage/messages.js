/*
 * FeaturePage Messages
 *
 * This contains all the text for the FeaturePage component.
 */
import {defineMessages} from 'react-intl'

export default defineMessages({
  cssHeader: {
    defaultMessage: 'Features',
    id: 'boilerplate.containers.FeaturePage.css.header'
  },
  cssMessage: {
    defaultMessage: 'Next generation CSS',
    id: 'boilerplate.containers.FeaturePage.css.message'
  },
  feedbackHeader: {
    defaultMessage: 'Instant feedback',
    id: 'boilerplate.containers.FeaturePage.feedback.header'
  },
  feedbackMessage: {
    defaultMessage: `
      Enjoy the best DX and code your app at the speed of thought! Your
    saved changes to the CSS and JS are reflected instantaneously
    without refreshing the page. Preserve application state even when
    you update something in the underlying code!
    `,
    id: 'boilerplate.containers.FeaturePage.feedback.message'
  },
  header: {
    defaultMessage: 'Features',
    id: 'boilerplate.containers.FeaturePage.header'
  },
  intlHeader: {
    defaultMessage:
      'Complete i18n Standard Internationalization & Pluralization',
    id: 'boilerplate.containers.FeaturePage.internationalization.header'
  },
  intlMessage: {
    defaultMessage:
      'Scalable apps need to support multiple languages, easily add and support multiple languages with `react-intl`.',
    id: 'boilerplate.containers.FeaturePage.internationalization.message'
  },
  javascriptHeader: {
    defaultMessage: 'Next generation JavaScript',
    id: 'boilerplate.containers.FeaturePage.javascript.header'
  },
  javascriptMessage: {
    defaultMessage: `Use template strings, object destructuring, arrow functions, JSX
    syntax and more, today.`,
    id: 'boilerplate.containers.FeaturePage.javascript.message'
  },
  networkHeader: {
    defaultMessage: 'Offline-first',
    id: 'boilerplate.containers.FeaturePage.network.header'
  },
  networkMessage: {
    defaultMessage: `
      The next frontier in performant web apps: availability without a
      network connection from the instant your users load the app.
    `,
    id: 'boilerplate.containers.FeaturePage.network.message'
  },
  routingHeader: {
    defaultMessage: 'Industry-standard routing',
    id: 'boilerplate.containers.FeaturePage.routing.header'
  },
  routingMessage: {
    defaultMessage: `
      Write composable CSS that's co-located with your components for
    complete modularity. Unique generated class names keep the
    specificity low while eliminating style clashes. Ship only the
    styles that are on the page for the best performance.
    `,
    id: 'boilerplate.containers.FeaturePage.routing.message'
  },
  scaffoldingHeader: {
    defaultMessage: 'Quick scaffolding',
    id: 'boilerplate.containers.FeaturePage.scaffolding.header'
  },
  scaffoldingMessage: {
    defaultMessage: `Automate the creation of components, containers, routes, selectors
  and sagas - and their tests - right from the CLI!`,
    id: 'boilerplate.containers.FeaturePage.scaffolding.message'
  },
  stateManagementHeader: {
    defaultMessage: 'Predictable state management',
    id: 'boilerplate.containers.FeaturePage.state_management.header'
  },
  stateManagementMessages: {
    defaultMessage: `
      Unidirectional data flow allows for change logging and time travel
    debugging.
    `,
    id: 'boilerplate.containers.FeaturePage.state_management.message'
  }
})
