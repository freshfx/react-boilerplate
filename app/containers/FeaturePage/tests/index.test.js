import React from 'react'
import {render} from '@testing-library/react'
import {HelmetProvider} from 'react-helmet-async'
import {IntlProvider} from 'react-intl'

import FeaturePage from '../index'

jest.mock('components/H1', () => props => <h1 {...props}>{props.children}</h1>)

const renderComponent = (props = {}) =>
  render(
    <IntlProvider locale="en">
      <HelmetProvider>
        <FeaturePage {...props} />
      </HelmetProvider>
    </IntlProvider>
  )

describe('<FeaturePage />', () => {
  it('should match snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })
})
