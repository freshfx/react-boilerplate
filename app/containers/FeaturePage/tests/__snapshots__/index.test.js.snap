// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturePage /> should match snapshot 1`] = `
.c0 {
  font-family: Georgia,Times,'Times New Roman',serif;
  padding-left: 1.75em;
}

.c1 {
  margin: 1em 0;
}

.c2 {
  font-weight: bold;
}

<div>
  <h1>
    Features
  </h1>
  <ul
    class="c0"
  >
    <li
      class="c1"
    >
      <p
        class="c2"
      >
        Quick scaffolding
      </p>
      <p>
        Automate the creation of components, containers, routes, selectors
  and sagas - and their tests - right from the CLI!
      </p>
    </li>
    s
    <li
      class="c1"
    >
      <p
        class="c2"
      >
        Industry-standard routing
      </p>
      <p>
        
      Write composable CSS that's co-located with your components for
    complete modularity. Unique generated class names keep the
    specificity low while eliminating style clashes. Ship only the
    styles that are on the page for the best performance.
    
      </p>
    </li>
    <li
      class="c1"
    >
      <p
        class="c2"
      >
        Offline-first
      </p>
      <p>
        
      The next frontier in performant web apps: availability without a
      network connection from the instant your users load the app.
    
      </p>
    </li>
    <li
      class="c1"
    >
      <p
        class="c2"
      >
        Complete i18n Standard Internationalization & Pluralization
      </p>
      <p>
        Scalable apps need to support multiple languages, easily add and support multiple languages with \`react-intl\`.
      </p>
    </li>
  </ul>
</div>
`;
